ngraph/core/reference/include/ngraph/runtime/reference/non_max_suppression.hpp

```diff
@@ -0,0 +1,81 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <functional>
+#include <map>
+#include <ngraph/runtime/host_tensor.hpp>
+#include <vector>
+#include "ngraph/node.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/ops.hpp"
+#include "ngraph/shape_util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            struct InfoForNMS5
+            {
+                int64_t max_output_boxes_per_class;
+                float iou_threshold;
+                float score_threshold;
+                float soft_nms_sigma;
+                Shape out_shape;
+                Shape boxes_shape;
+                Shape scores_shape;
+                std::vector<float> boxes_data;
+                std::vector<float> scores_data;
+                size_t out_shape_size;
+                bool sort_result_descending;
+                ngraph::element::Type output_type;
+            };
+
+            InfoForNMS5 get_info_for_nms5_evaluation(const op::v5::NonMaxSuppression* nms5,
```

Comment:
It is a service method I propose to remove this method from the file with reference implementation and move to the place where we call the reference implementation (to Interpreter backend)